### PR TITLE
Be stricter about invalid UTF-8 input

### PR DESCRIPTION
**Pull request by neilmayhew**
Keep track of whether unicode strings were created from valid UTF-8 and use this to abort operations that use these strings later.

Typically, this is done by returning an empty string when a modified string is being produced, since an empty string is never a valid phone number.

See [b/206677455](https://issuetracker.google.com/issues/206677455)

---
This change is [Reviewable](https://reviewable.io/reviews/google/libphonenumber/2698)

### CODE REVIEW

File: cpp/src/phonenumbers/normalize_utf8.h [hunk 1]
```diff
@@ -27,6 +27,8 @@ struct NormalizeUTF8 {
     string normalized;
     UnicodeText number_as_unicode;
     number_as_unicode.PointToUTF8(number.data(), static_cast<int>(number.size()));
+    if (!number_as_unicode.UTF8WasValid())
+      return normalized;
```
Comment (katbohm) [inline cpp/src/phonenumbers/normalize_utf8.h:31]: Maybe add a comment where normalized is declared (or here) to state that it is initialized to an empty string and this value is intended to be returned here. Also, do we have a test case covering that a non-utf8 string will be normalized to empty string?

Comment (neilmayhew) [reply cpp/src/phonenumbers/normalize_utf8.h]: Done

**Comment by penmetsaa** [comment]
Hi,

Thanks so much for the updates. The changes looks good for me.

Could you please update the branch with master merge.
So we can approve and merge?

**Comment by penmetsaa** [comment]
Hi,
Could you please update the branch with master merge.
So we can approve and merge?

**Comment by neilmayhew** [comment]
I rebased and squashed.

**Review by penmetsaa — APPROVED** [review]

**Comment by neilmayhew** [comment]
I'm surprised you didn't mention this change in the release notes. However, it probably won't affect many people.
